txgossip/pushpull.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txgossip
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/network/p2p/gossip"
+	"github.com/ava-labs/libevm/core/txpool"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/libevm/ethapi"
+)
+
+// SendTx implements the respective method of [ethapi.Backend], accepting
+// transactions submitted via the `eth_sendTransaction` RPC method. Unlike
+// [gossip.BloomSet.Add], transactions added via this method will be
+// push-gossiped; see [Set.RegisterPushGossiper].
+func (s *Set) SendTx(ctx context.Context, ethTx *types.Transaction) error {
+	var _ ethapi.Backend // protect the import for [comment] rendering
+
+	// We filter out [txpool.ErrAlreadyKnown] in [txSet.Add] but want to keep it
+	// for RPCs.
+	if s.set.pool.Has(ethTx.Hash()) {
+		return txpool.ErrAlreadyKnown
+	}
+
+	tx := Transaction{ethTx}
+	if err := s.BloomSet.Add(tx); err != nil {
+		return err
+	}
```

Comment:
Commenting on a behavior change here: normally we would mark these transactions as local on the API side. But this doesn't mark these as local txs.

I don't _think_ this is a problem... But figured it is worth a comment.

Typically (for an API node), I don't think we'd actually want to specify something as local... Since that permanently marks the account as local... But some people may want that behavior.